storage/jpa/src/main/java/io/apicurio/registry/storage/impl/jpa/JPARegistryStorage.java

```diff
@@ -95,6 +95,7 @@
     @Inject
     ArtifactTypeUtilProviderFactory factory;
 
+    // TODO Could there be a race condition here? The new max+1 version is saved with a new artifact
```

Comment:
It looks like it could be, unless you have some constraint on the tuple (<aritfactId, version>) ?
Then you must handle this constraint in a loop -- allowing it to fail, and get next version, ...